include/xtensor/xfixed.hpp

```diff
@@ -0,0 +1,577 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay and Wolf Vollprecht    *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_FIXED_HPP
+#define XTENSOR_FIXED_HPP
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <utility>
+#include <vector>
+
+#include "xcontainer.hpp"
+#include "xsemantic.hpp"
+
+namespace xt
+{
+
+    // Fixed shape
+    template <class T, std::size_t N>
+    struct const_array
+    {
+        using size_type = std::size_t;
+        using value_type = T;
+        using pointer = value_type*;
+        using const_pointer = const value_type*;
+        using reference = value_type&;
+        using const_reference = const value_type&;
+
+        using iterator = pointer;
+        using const_iterator = const_pointer;
+
+        constexpr std::size_t operator[](std::size_t idx) const
+        {
+            return m_data[idx];
+        }
+
+        constexpr const_iterator begin() const
+        {
+            return m_data;
+        }
+
+        constexpr const_iterator end() const
+        {
+            return m_data + N;
+        }
+
+        constexpr const_iterator cbegin() const
+        {
+            return begin();
+        }
+
+        constexpr const_iterator cend() const
+        {
+            return end();
+        }
+
+        constexpr const_reference front() const
+        {
+            return m_data[0];
+        }
+
+        constexpr const_reference back() const
+        {
+            return m_data[size() - 1];
+        }
+
+        constexpr std::size_t size() const
+        {
+            return N;
+        }
+
```

Comment:
It would be good to implement `erase()` and `insert()` etc. such that they return a new array with the appropriate modifications. I imagine this to be the generic behavior of `tiny_array`-like classes - they should not in general be able to modify their size, but can return a resized copy.